api/illumio/cloud/k8sclustersync/v1/k8s_info.proto

```diff
@@ -87,6 +88,73 @@ message KubernetesPodData {
   repeated string ip_addresses = 1;
 }
 
+// KubernetesNetworkPolicyData contains information specific to a NetworkPolicy object.
+message KubernetesNetworkPolicyData {
+  bool ingress = 1; // True if Ingress policy is enabled
+  bool egress = 2; // True if Egress policy is enabled
+  LabelSelector pod_selector = 3; // Pod selector for the policy
+  repeated NetworkPolicyRule ingress_rules = 4; // Ingress rules for the policy
+  repeated NetworkPolicyRule egress_rules = 5; // Egress rules for the policy
+}
+
+// LabelSelector selects a k8s resource (e.g., a pod or namespace) by its labels.
+message LabelSelector {
+  map<string, string> match_labels = 1;
+  repeated LabelSelectorRequirement match_expressions = 2;
+}
+
+// LabelSelectorRequirement is a selector that contains values, a key, and an operator that relates the key and values.
+message LabelSelectorRequirement {
+  string key = 1;
+  // operator represents a key's relationship to a set of values. Valid operators are In, NotIn, Exists and DoesNotExist.
+  string operator = 2;
+  repeated string values = 3;
+}
+
+// IPBlock defines a particular CIDR (Ex: 10.10.10.0/24).
+message IPBlock {
+  string cidr = 1;
+  repeated string except = 2;
+}
+
+// Peer defines a network peer as a set of IP addresses or a set of pods.
+message Peer {
+  oneof peer {
+    // Defines policy on a particular IPBlock. If this field is set, neither of the other fields can be.
```

Comment:
Nit. The `oneof` makes it already clear that only one of `ip_block` or `pod` can be set.
```suggestion
```